### PR TITLE
Fix unit tests

### PR DESCRIPTION
**Pull request by pandamicro**
Re: cocos-creator/fireball#4330

Changes proposed in this pull request:
 * Fix unit test issues caused by color
 * Fix loader & pipeline unit test issues

> Makes sure these boxes are checked before submitting your PR - thank you!
>
- [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
- [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
- To official teams:
  - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
  - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
  - [x] Make sure any log information in `cc.log` , `cc.error`, `cc.warn` or `cc.assert` has been moved into `DebugInfos.js` with an ID

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/value-types/CCColor.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ var Color = (function () {
         g = g || 0;
         b = b || 0;
         a = typeof a === 'number' ? a : 255;
-        this._val = ((r << 24) >>> 0) + (g << 16) + (b << 8) + a;
+        this._val = ((~~r << 24) >>> 0) + (~~g << 16) + (~~b << 8) + ~~a;
```
Comment (jareguo) [inline cocos2d/core/value-types/CCColor.js:68]: `~~`是取整吗…… 统一用 `| 0` 吧？和 asm.js 保持一致

File: cocos2d/core/value-types/CCColor.js [hunk 2]
```diff
@@ -310,7 +310,7 @@ var Color = (function () {
      * color.setB(255); // Color {r: 0, g: 0, b: 255, a: 255}
      */
     proto.setB = function (blue) {
-        this._val = (this._val & 0xffff00ff) | (blue << 8);
+        this._val = ((this._val & 0xffff00ff) | (~~blue << 8)) >>> 0;
```
Comment (jareguo) [inline cocos2d/core/value-types/CCColor.js:313]: 右移 0 位是做什么用的？？

Comment (pandamicro) [reply cocos2d/core/value-types/CCColor.js:313]: 保持这个值是 unsigned int，否则会自动变成 signed

File: cocos2d/core/value-types/CCColor.js (hunk 1 above)
Comment (pandamicro) [reply cocos2d/core/value-types/CCColor.js:68]: 不要。。。因为 | 运算符的优先级低于 <<，我得再写好多个括号。。。

Comment (jareguo) [reply cocos2d/core/value-types/CCColor.js:68]: 机智如你~

**Review by jareguo — APPROVED** [review]

**Pull request merged**